src/AutoRest.CSharp.V3/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -70,6 +65,16 @@ public async IAsyncEnumerable<(string Name, string Text)> GetGeneratedFilesAsync
 
         private static async Task<Document> ProcessDocument(Document document)
         {
+            var compilation = document.Project.GetCompilationAsync().GetAwaiter().GetResult();
```

Comment:
Now that you're in an async method, might as well use `await` for cleaner code.

Reply:
Good point.